homeassistant/components/airvisual/__init__.py

```diff
@@ -343,10 +345,7 @@ async def async_unload_entry(hass, config_entry):
         remove_listener = hass.data[DOMAIN][DATA_LISTENER].pop(config_entry.entry_id)
         remove_listener()
 
-        if (
```

Comment:
Oooh, great catch! If someone added a name-based entry, this would not have happened previously, which is definitely a bug. 👍🏻